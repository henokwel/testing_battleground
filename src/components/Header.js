import React from 'react'

export const Header = () => {
    return (
        <header className="App-header">
            <h1>Header</h1>
        </header>
    )
}
