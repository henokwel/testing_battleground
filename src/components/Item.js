import React, { useState } from 'react'

export const Item = ({ id, title, done }) => {
    const [edit, setEdit] = useState(false)

    return (
        <div key={id} className='item'>
            {
                !edit ?
                    <p>{title}</p>
                    :
                    <input type="text" value={title} />
            }

            <div className='item_btn_Container'>
                <input type="checkbox" value={done} />
                <button onClick={() => setEdit(!edit)}>{edit ? "Done" : "Edit"}</button>
            </div>

        </div>
    )
}
