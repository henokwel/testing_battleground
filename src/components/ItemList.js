import React from 'react'
import { Item } from '../components/Item'
export const ItemList = ({data}) => {
    console.log(data);
    
    return (
        <div className='itemList'>

            {
                !data? <p>No Data</p> :
                    data.map(item => <Item {...item} />)
            }
        </div>
    )
}
