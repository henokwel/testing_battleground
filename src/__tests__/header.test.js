
import { shallow } from "enzyme";
import { Header } from "../components/Header";


describe('Header', () => {
    test('should render Header ', () => {

        const wrapper = shallow(<Header />)
        expect(wrapper).toMatchSnapshot()
    })

    test('should render correct text', () => {
        const wrapper = shallow(<Header />)
        expect(wrapper.find("h1").text()).toBe("Header")
    })
})



