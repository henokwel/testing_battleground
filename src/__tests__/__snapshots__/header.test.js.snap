// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header  1`] = `
<header
  className="App-header"
>
  <h1>
    Header
  </h1>
</header>
`;
